File: app/src/main/java/com/prateek/dogengine/data/BreedImageDataSourceFactory.java
package com.prateek.dogengine.data;

import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;

public class BreedImageDataSourceFactory extends DataSource.Factory<Integer, BreedImage> {
    private BreedImageDataSource breedImageDataSource;
    private MutableLiveData<BreedImageDataSource> liveData;
    private int mBreedId;

    public BreedImageDataSourceFactory(int breedId) {
        liveData = new MutableLiveData<>();
        mBreedId = breedId;
    }

    @Override
    public DataSource create() {
        breedImageDataSource = new BreedImageDataSource(mBreedId);
        liveData.postValue(breedImageDataSource);
        return breedImageDataSource;
    }
}

File: app/src/main/kotlin/com/prateek/dogengine/data/BreedImageDataSourceFactory.kt
package com.prateek.dogengine.data

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource

class BreedImageDataSourceFactory(private val breedId: Int) :
    DataSource.Factory<Int, BreedImage>() {
    private var breedImageDataSource: BreedImageDataSource? = null
    private val liveData = MutableLiveData<BreedImageDataSource>()

    override fun create(): DataSource<Int?, BreedImage?> {
        breedImageDataSource = BreedImageDataSource(breedId)
        liveData.postValue(breedImageDataSource)
        return breedImageDataSource!!
    }
}